### PR TITLE
Add section about `lib` function visibility in macro

### PR DESCRIPTION
**Pull request by straight-shoota**
cf. https://github.com/crystal-lang/crystal/pull/12848

### CODE REVIEW

File: docs/syntax_and_semantics/c_bindings/lib.md [hunk 1]
```diff
@@ -17,3 +17,14 @@ Attributes are used to pass flags to the linker to find external libraries:
...
+
+## Reflection
+
+Lib functions are visible in the macro language anywhere in the program.
+
+```crystal
+lib LibFoo
+  fun foo
+end
+{% if LibFoo.methods %} # => [fun foo]
```
Comment (Blacksmoke16) [inline docs/syntax_and_semantics/c_bindings/lib.md:29]:
```suggestion
{{ LibFoo.methods }} # => [fun foo]
```

File: docs/syntax_and_semantics/c_bindings/lib.md [hunk 2]
```diff
@@ -17,3 +17,14 @@ Attributes are used to pass flags to the linker to find external libraries:
 * `@[Link(framework: "Cocoa")]` will pass `-framework Cocoa` to the linker (only useful in macOS).
 
 Attributes can be omitted if the library is implicitly linked, as in the case of libc.
+
+## Reflection
+
+Lib functions are visible in the macro language anywhere in the program.
```
Comment (beta-ziliani) [inline docs/syntax_and_semantics/c_bindings/lib.md:23]: I don't think it's clear what "anywhere" means here. Also, probably we want to link to the API.

```suggestion
Lib functions are visible in the macro language using the method [`TypeNode#methods`](https://crystal-lang.org/api/1.6.2/Crystal/Macros/TypeNode.html#methods%3AArrayLiteral%28Def%29-instance-method):
```

**Review by beta-ziliani — COMMENTED** [review]
It's a bit odd that we're saying this about libs, but not about other types. But I guess it's better to have something somewhere than anything, so OK.

File: docs/syntax_and_semantics/c_bindings/lib.md [hunk 3]
```diff
@@ -17,3 +17,14 @@ Attributes are used to pass flags to the linker to find external libraries:
...
+
+```crystal
+lib LibFoo
+  fun foo
+end
+{{ LibFoo.methods }} # => [fun foo]
```
Comment (beta-ziliani) [inline docs/syntax_and_semantics/c_bindings/lib.md:29]: Formatting is complaining
```suggestion

{{ LibFoo.methods }} # => [fun foo]
```

File: docs/syntax_and_semantics/c_bindings/lib.md (hunk 2 above)
Comment (straight-shoota) [reply docs/syntax_and_semantics/c_bindings/lib.md:23]: "anywhere" might not be very clear, but it's important. `TypeNode#methods` for other types only works within method context. The point we're mentioning this here is that for lib types it also works outside of methods.

**Review by beta-ziliani — APPROVED** [review]

File: docs/syntax_and_semantics/c_bindings/lib.md (hunk 2 above)
Comment (beta-ziliani) [reply docs/syntax_and_semantics/c_bindings/lib.md]: but if it's important, it should be clear. anyway, we can leave it for a future improvement.

**Pull request merged**